src/fq.rs

```diff
@@ -619,3 +666,15 @@ fn test_inversion() {
         tmp.add_assign(&R2);
     }
 }
+
+#[test]
+fn test_square_assign_equals_mul_assign() {
```

Comment:
I think this test is better off as a black-box test later; the existing test for squaring should handle this.